src/UNL/Peoplefinder/Developers/AbstractResource.php

```diff
@@ -0,0 +1,53 @@
+<?php
+abstract class UNL_PeopleFinder_Developers_AbstractResource
+{
+    const FORMAT_JSON = 'json';
+    const FORMAT_XML = 'xml';
+    const FORMAT_PARTIAL = 'partial';
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getTitle();
+    
+    /**
+     * @return string - a brief description of the resource
+     */
+    abstract public function getDescription();
+
+    /**
+     * @return mixed - an associative array of property=>description
+     */
+    abstract public function getAvailableFormats();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getJsonProperties();
+
+    /**
+     * @return array - an associative array of property=>description
+     */
+    abstract public function getXmlProperties();
```

Comment:
I'll accept that for now, but the arguments about the differences between the formats is not a valid one. Data is data. It shouldn't matter that JSON uses arrays and XML uses elements, etc.